pandas/tests/frame/methods/test_join.py

```diff
@@ -104,6 +107,126 @@ def test_suffix_on_list_join():
     tm.assert_frame_equal(arr_joined, norm_joined)
 
 
+def test_join_on_single_col_check_dup():
+    # GH 46622
```

Comment:
Could you either try parametrizing this or creating multiple tests? This is hard to debug if something goes wrong

Reply:
If this test fails, it's easy to trace the issue with the line and corresponding comments in this function. I don't think of an easy way of parametrizing that does not make this messier. Could you double check?

Reply:
Multiple tests are fine too. 

If the first fails all other checks are not executed. Also it is not easy to see how the df looks if a check further below fails

Reply:
Just changed. Does it look better now? Thanks.

Reply:
This is still one big test?

Reply:
Oh sorry. Forgot to add test_join.py after resolving conflict with the upstream/main. Please check again